Letterbook.Core.Tests/TheoryData.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+
+namespace Letterbook.Core.Tests;
+
+/// <summary>
+/// A support class that allows us to have more idiomatic and type-safe ClassData classes to drive XUnit Theory tests
+/// ClassData implementations should inherit one of the generic derivatives of this class.
+/// </summary>
+public abstract class TheoryData : IEnumerable<object[]>
```

Comment:
TheoryData is a test utility that lets us have explicitly typed ClassData providers for XUnit theories.